cunumeric/array.py

```diff
@@ -237,6 +406,13 @@ def __init__(
         else:
             self._thunk = thunk
         self._legate_data: Union[dict[str, Any], None] = None
+        # TODO: flags are passed. We use this argument only for views
```

Comment:
We may want to do the reverse as well, i.e. propagate the flags when we create a NumPy array backed by a cuNumeric array. Specifically, I would suggest adding a `flags` argument to `NumpyThunk.__numpy_array__()`, and pass the flags from `self` when calling `__numpy_array__` from `cunumeric.ndarray.__array__()`.